sktime/forecasting/chronos.py

```diff
@@ -0,0 +1,161 @@
+# !/usr/bin/env python3 -u
+# copyright: sktime developers, BSD-3-Clause License (see LICENSE file)
+"""Implements Chronos forecaster by wrapping amazon's chronos."""
+
+import numpy as np
+import pandas as pd
+
+__author__ = ["RigvedManoj"]
+__all__ = ["Chronos"]
+
+
+from sktime.forecasting.base import BaseForecaster
+
+
+# todo: add any necessary imports here
+
+
+class Chronos(BaseForecaster):
+    """Chronos forecaster by wrapping Amazon's Chronos model [1]_.
+
+    Direct interface to Amazon Chronos, using the sktime interface.
+    All hyperparameters are exposed via the constructor.
+
+    Parameters
+    ----------
+    modelName: str, required
+    top_p: float, default=1.0
+    top_k: int, default=50
+    temperature: float, default=1.0
+    num_samples: int, default=20
+    args_list: list, default=None
+    kwargs_dict: dict, default=None
+
+    References
+    ----------
+    . [1] https://github.com/amazon-science/chronos-forecasting
+
+    Examples
+    --------
+    >>> from sktime.datasets import load_airline
+    >>> from sktime.forecasting.chronos import Chronos
+    >>> from sktime.split import temporal_train_test_split
+    >>> from sktime.performance_metrics.forecasting import mean_absolute_percentage_error
+    >>> from sktime.forecasting.base import ForecastingHorizon
+    >>> import torch
+    >>> y = load_airline()
+    >>> y_train, y_test = temporal_train_test_split(y)
+    >>> fh = ForecastingHorizon(y_test.index, is_relative=False)
+    >>> forecaster = Chronos("amazon/chronos-t5-small", kwargs_dict={"torch_dtype": torch.bfloat16})
+    >>> forecaster.fit(y_train)
+    >>> y_pred = forecaster.predict(fh)
+    >>> mean_absolute_percentage_error(y_test, y_pred)
+    """
+
+    # tag values are "safe defaults" which can usually be left as-is
+    _tags = {
+        "python_dependencies": ["torch"],
+        "y_inner_mtype": "pd.Series",
+        "X_inner_mtype": "pd.DataFrame",
+        "scitype:y": "univariate",
+        "ignores-exogeneous-X": True,
+        "requires-fh-in-fit": False,
+        "authors": ["RigvedManoj"],
+        "maintainers": ["RigvedManoj"],
+    }
+
+    def __init__(self, modelName, num_samples=20, temperature=1.0, top_k=50, top_p=1.0, args_list=None, kwargs_dict=None):
```

Comment:
How do I interpret temperature? In a text generation model, I can understand where sometimes we may need creativity. In this context, what will 1 (max) and 0 (min) imply?

Reply:
I think I'll have similar doubts for top-p and top-k as well.